test/e2e/delete_test.go

```diff
@@ -276,6 +270,45 @@ func (tc *testContext) testWindowsNodeDeletion(t *testing.T) {
 	require.NoError(t, err, "could not delete test namespace")
 }
 
+// deployWorkloadsToDrain creates workloads that require special options to drain
+func (tc *testContext) deployWorkloadsToDrain() error {
+	// Deploy a DaemonSet and wait until its pods have been made ready on each Windows node. DaemonSet pods cannot be
+	// drained from a Node.
+	// Doing this ensures that WMCO is able to handle containers being present on the instance when deconfiguring.
+	ds, err := tc.deployNOOPDaemonSet()
+	if err != nil {
+		return fmt.Errorf("error creating DaemonSet %w", err)
+	}
+	defer tc.client.K8s.AppsV1().DaemonSets(ds.GetNamespace()).Delete(context.TODO(), ds.GetName(),
+		meta.DeleteOptions{})
+	err = tc.waitUntilDaemonsetScaled(ds.GetName(), int(gc.numberOfMachineNodes+gc.numberOfBYOHNodes))
```

Comment:
With the proposed refactoring of the[ deployment code](https://github.com/openshift/windows-machine-config-operator/pull/1887/files#diff-cd50fad6da0839c40258ca356bf995a7ee5e1d5c270ee0af2cc5752e2f10a454L212-L220) to another function, the workload deletion test loses its value, i.e. the presence of workloads during the deletion. 

Both deployments are getting deleted right after the `deployWorkloadsToDrain()`  function returns, leaving no running workloads by the [time the nodes are getting drained](https://github.com/openshift/windows-machine-config-operator/pull/1887/files#diff-cd50fad6da0839c40258ca356bf995a7ee5e1d5c270ee0af2cc5752e2f10a454R231-R242). 

See the [WMCO log in one of the jobs that passed in this PR](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/1887/pull-ci-openshift-windows-machine-config-operator-master-aws-e2e-upgrade/1714735679407132672/artifacts/aws-e2e-upgrade/e2e-aws-upgrade/artifacts/wmco.log) and compare it with[ WMCO log from an older job](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/1834/pull-ci-openshift-windows-machine-config-operator-master-aws-e2e-upgrade/1714313236225986560/artifacts/aws-e2e-upgrade/e2e-aws-upgrade/artifacts/wmco.log) where there is evidence of the eviction of the pods. 


````
2023-10-17T18:32:55Z	INFO	nc 10.0.35.103	deconfiguring
2023-10-17T18:32:55Z	ERROR	nc 10.0.35.103	WARNING: ignoring DaemonSet-managed Pods: wmco-test/noop-ds-vbwvb
````


I recommend deploying the jobs in separate functions and returning the job names to the parent execution block to delete them after the machine and BYOH instances are deleted/deconfigured. Another option is to return a cleanup function aware of the jobs and defer its execution from the parent thread.